spec/rubocop/cop/style/combinable_defined_spec.rb

```diff
@@ -0,0 +1,144 @@
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(::Foo::Bar)
+      RUBY
+    end
+
+    it 'registers an offense for three `defined?`s with same nesting' do
+      expect_offense(<<~RUBY)
+        defined?(Foo) && defined?(Foo::Bar) && defined?(Foo::Bar::Baz)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(Foo::Bar::Baz)
+      RUBY
+    end
+
+    it 'does not register an offense for two `defined?`s with same namespace but different nesting' do
+      expect_no_offenses(<<~RUBY)
+        defined?(Foo::Bar) && defined?(Foo::Baz)
+      RUBY
+    end
+
+    it 'does not register an offense for two `defined?`s with negation' do
+      expect_no_offenses(<<~RUBY)
+        defined?(Foo) && !defined?(Foo::Bar)
+      RUBY
+    end
+
+    it 'does not register an offense for two `defined?` with different `cbase`s' do
+      expect_no_offenses(<<~RUBY)
+        defined?(::Foo) && defined?(Foo::Bar)
+      RUBY
+    end
+
+    it 'does not register an offense when skipping a nesting level' do
+      expect_no_offenses(<<~RUBY)
+        defined?(Foo) && defined?(Foo::Bar::Baz)
+      RUBY
+    end
+
+    it 'registers an offense when the namespace is not a constant' do
+      expect_offense(<<~RUBY)
+        defined?(foo) && defined?(foo::Bar)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(foo::Bar)
+      RUBY
+    end
+
+    it 'registers an offense for method chain with dots' do
+      expect_offense(<<~RUBY)
+        defined?(foo) && defined?(foo.bar)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(foo.bar)
+      RUBY
+    end
+
+    it 'registers an offense for method chain with `::`' do
+      expect_offense(<<~RUBY)
+        defined?(foo) && defined?(foo::bar)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(foo::bar)
+      RUBY
+    end
+
+    it 'registers an offense for a method chain followed by constant nesting' do
+      expect_offense(<<~RUBY)
+        defined?(foo) && defined?(foo.bar) && defined?(foo.bar::BAZ)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(foo.bar::BAZ)
+      RUBY
+    end
+
+    it 'registers an offense when there is another term between `defined?`s' do
+      expect_offense(<<~RUBY)
+        foo && defined?(Foo) && bar && defined?(Foo::Bar)
```

Comment:
Seems like this would cause `bar` to run after correction where it would previously not, if `Foo` or `Foo::Bar` is not defined.

I don't think it should register an offense if there are terms in-between.

Reply:
Fair point. I'll change to not register if there are extra terms.